experiments/sandbox-next/src/utils/codegen/core/codegen.ts

```diff
@@ -0,0 +1,182 @@
+    const components: Record<string, ComponentValidationInfo> = {};
+
+    for (const component of scanResult.components) {
+      const issues = this.generateComponentIssues(component);
+      const { status } = this.determineComponentStatus(issues);
+
+      components[component.name] = {
+        path: component.componentPath,
+        status,
+        issues,
+      };
+    }
+
+    return {
+      timestamp: new Date().toISOString(),
+      components,
+      summary: this.generateSummary(components),
+    };
+  }
+
+  private generateComponentIssues(component: ComponentInfo): ValidationIssue[] {
+    const issues: ValidationIssue[] = [];
+
+    if (component.validation?.componentValidation) {
+      const validation = component.validation.componentValidation;
+      if (!validation.hasDefaultExport) {
+        issues.push({
+          file: "index",
+          type: "missing_export",
+          severity: "warn",
+          message: `Missing default export in index.tsx`,
+        });
+      }
+    }
+
+    if (component.validation?.transformerValidation) {
+      const validation = component.validation.transformerValidation;
+      if (!validation.hasDefaultExport) {
+        issues.push({
+          file: "transformer",
+          type: "missing_export",
+          severity: "warn",
+          message: `Missing default export in transformer`,
+        });
+      }
+    }
+
+    return issues;
+  }
+
+  private determineComponentStatus(issues: ValidationIssue[]): {
+    status: "valid" | "warning" | "error";
+  } {
+    if (issues.some((i) => i.severity === "error")) {
+      return { status: "error" };
+    }
+    if (issues.some((i) => i.severity === "warn")) {
+      return { status: "warning" };
+    }
+    return { status: "valid" };
+  }
+
+  private generateSummary(
+    components: Record<string, ComponentValidationInfo>
+  ): ValidationReport["summary"] {
+    return Object.values(components).reduce(
+      (acc, { status }) => {
+        if (status === "valid") acc.valid++;
+        else if (status === "warning") acc.withWarnings++;
+        else acc.withErrors++;
+        return acc;
+      },
+      {
+        total: Object.keys(components).length,
+        valid: 0,
+        withWarnings: 0,
+        withErrors: 0,
+      }
+    );
+  }
+
+  async generate(): Promise<void> {
+    console.log("🔍 Scanning component directory...");
+
+    try {
+      const scanResult = await this.getScanResult(true); // Force fresh scan
+      const template = new Template(scanResult, this.config);
+
+      await ensureDirectoryExists(this.config.outputFile);
+      await writeFile(this.config.outputFile, template.toString(), "utf-8");
+
+      console.log("✨ Component registry generated successfully");
+    } catch (error) {
+      this.handleError(
+        error,
+        CodegenErrorCode.TEMPLATE_GENERATION_FAILED,
+        "Failed to generate component registry"
+      );
+    }
+  }
+
+  async validate(): Promise<ValidationReport> {
```

Comment:
this returns a report, maybe a better name method could be generateReport or getReport